### PR TITLE
Issue 6820 - etc.c.curl missing const

### PR DESCRIPTION
**Pull request by RazvanN7**
Some of the attributes of curl functions were missing the const qualifier.

### CODE REVIEW

File: etc/c/curl.d [hunk 1]
```diff
@@ -1601,7 +1601,7 @@ struct curl_slist
  * Appends a string to a linked list. If no list exists, it will be created
  * first. Returns the new list, after appending.
  */
-curl_slist * curl_slist_append(curl_slist *, in char *);
```
Comment (UplinkCoder) [inline etc/c/curl.d]: In is equivalent to const

Comment (RazvanN7) [inline etc/c/curl.d]: I though it would be more consistent with the original implementation

Comment (RazvanN7) [inline etc/c/curl.d]: should I replace them back with in? I thought it would be a more precise consistency if we used just const

**Comment by mathias-lang-sociomantic** [comment]
You cannot *just* change the callback types.
Because delegates have no parameter contravariance, it's not possible to pass a `void delegate (const(char)* foo)` instead of a `void delegate(char*)`. Also, some people might be relying on this exact signature.

You have to:
- Introduce new names for those symbols, with a new signature, as completely new symbols
- Add overloads of anything that could use those callbacks to accept the new callbacks
- Deprecate the old callback types, and everything using it, alltogether.

I suggest to do it in separate commit, so e.g. if a deprecation proves problematic it can be reverted without reverting the new symbols addition.

**Comment by andralex** [comment]
Hmm, adding @MartinNowak to decide the best path here. Basically the old code was in error:

* https://github.com/curl/curl/blob/master/include/curl/curl.h#L286: curl_chunk_bgn_callback takes a const in the first position

* https://github.com/curl/curl/blob/master/include/curl/curl.h#L733: curl_sshkeycallback takes two consts

* https://github.com/curl/curl/blob/master/include/curl/curl.h#L2187: curl_getdate takes two pointers to const

Of course the nicest solution would be to allow contravariance of parameter types for delegates at least in the case of const. @WalterBright how difficult would that be? Shall we bootcamp that instead?

**Comment by mathias-lang-sociomantic** [comment]
> Of course the nicest solution would be to allow contravariance of parameter types for delegates at least in the case of const. @WalterBright how difficult would that be? Shall we bootcamp that instead?

Yes please !
See https://issues.dlang.org/show_bug.cgi?id=16303 and the linked entry in the first comment.

**Comment by andralex** [comment]
I'll close this.

**Pull request closed without merging**